### DISCUSSION TITLE
Connection reset by peer (104)

### DISCUSSION THREAD

**Discussion by PedroCoelhho** [epoptes/epoptes#99] [Q&A]
When i conect from my cliente on server, his disconect and show on the log this mensage: main: read: Connection reset by peer (104).
I need little help, because i don't understand what the connection keeps falling.
Server: Ubuntu 18.04
* Epoptes was instaled on package ltsp-server-standalone

**Comment by alkisg** [comment]
What is the output of this command, both on the server and the client?

```shell
dpkg-query -W epoptes epoptes-client
```

Also, which desktop environment are you running, e.g. MATE, GNOME, KDE?

Finally, are you using thin or fat LTSP clients?

**Comment by PedroCoelhho** [comment]
command exit:
dpkg-query -W epoptes epoptes-client

Client
epoptes	1.0.1-2+t201812111158~ubuntu18.04.1
epoptes-client	1.0.1-2+t201812111158~ubuntu18.04.1


Server
epoptes	1.0.1-2+t201812111158~ubuntu18.04.1
epoptes-client	1.0.1-2+t201812111158~ubuntu18.04.1

MATE environment
FAT CLIENTS

**Comment by alkisg** [comment]
Open a terminal on the server, and run:

```shell
epoptes
```

Then open a terminal on a client, and run:

```shell
/usr/sbin/epoptes-client
```

Wait until the problem happens, then copy/paste all the output of both terminals here.

**Comment by PedroCoelhho** [comment]
Enswer Client

atendimento7@ltsp230:~$ /usr/sbin/epoptes-client 
Epoptes-client connecting to server:789...   ...done
###############################################################
#@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@#
#@                                                           @#
#@  **  WARNING  **  WARNING  **  WARNING  **  WARNING  **   @#
#@                                                           @#
#@        YOU ARE RUNNING X11VNC WITHOUT A PASSWORD!!        @#
#@                                                           @#
#@  This means anyone with network access to this computer   @#
#@  may be able to view and control your desktop.            @#
#@                                                           @#
#@ >>> If you did not mean to do this Press CTRL-C now!! <<< @#
#@                                                           @#
#@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@#
#@                                                           @#
#@  You can create an x11vnc password file by running:       @#
#@                                                           @#
#@       x11vnc -storepasswd password /path/to/passfile      @#
#@  or   x11vnc -storepasswd /path/to/passfile               @#
#@  or   x11vnc -storepasswd                                 @#
#@                                                           @#
#@  (the last one will use ~/.vnc/passwd)                    @#
#@                                                           @#
#@  and then starting x11vnc via:                            @#
#@                                                           @#
#@      x11vnc -rfbauth /path/to/passfile                    @#
#@                                                           @#
#@  an existing ~/.vnc/passwd file from another VNC          @#
#@  application will work fine too.                          @#
#@                                                           @#
#@  You can also use the -passwdfile or -passwd options.     @#
#@  (note -passwd is unsafe if local users are not trusted)  @#
#@                                                           @#
#@  Make sure any -rfbauth and -passwdfile password files    @#
#@  cannot be read by untrusted users.                       @#
#@                                                           @#
#@  Use x11vnc -usepw to automatically use your              @#
#@  ~/.vnc/passwd or ~/.vnc/passwdfile password files.       @#
#@  (and prompt you to create ~/.vnc/passwd if neither       @#
#@  file exists.)  Under -usepw, x11vnc will exit if it      @#
#@  cannot find a password to use.                           @#
#@                                                           @#
#@                                                           @#
#@  Even with a password, the subsequent VNC traffic is      @#
#@  sent in the clear.  Consider tunnelling via ssh(1):      @#
#@                                                           @#
#@    http://www.karlrunge.com/x11vnc/#tunnelling            @#
#@                                                           @#
#@  Or using the x11vnc SSL options: -ssl and -stunnel       @#
#@                                                           @#
#@  Please Read the documention for more info about          @#
#@  passwords, security, and encryption.                     @#
#@                                                           @#
#@    http://www.karlrunge.com/x11vnc/faq.html#faq-passwd    @#
#@                                                           @#
#@  To disable this warning use the -nopw option, or put     @#
#@  'nopw' on a line in your ~/.x11vncrc file.               @#
#@                                                           @#
#@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@#
###############################################################
06/08/2019 13:14:26 x11vnc version: 0.9.13 lastmod: 2011-08-10  pid: 8510
06/08/2019 13:14:26 Using X display :7.0
06/08/2019 13:14:26 rootwin: 0x106 reswin: 0x2600001 dpy: 0xff3cfcd0
06/08/2019 13:14:26 
06/08/2019 13:14:26 ------------------ USEFUL INFORMATION ------------------
06/08/2019 13:14:26 X DAMAGE available on display, using it for polling hints.
06/08/2019 13:14:26   To disable this behavior use: '-noxdamage'
06/08/2019 13:14:26 
06/08/2019 13:14:26   Most compositing window managers like 'compiz' or 'beryl'
06/08/2019 13:14:26   cause X DAMAGE to fail, and so you may not see any screen
06/08/2019 13:14:26   updates via VNC.  Either disable 'compiz' (recommended) or
06/08/2019 13:14:26   supply the x11vnc '-noxdamage' command line option.
06/08/2019 13:14:26 
06/08/2019 13:14:26 Wireframing: -wireframe mode is in effect for window moves.
06/08/2019 13:14:26   If this yields undesired behavior (poor response, painting
06/08/2019 13:14:26   errors, etc) it may be disabled:
06/08/2019 13:14:26    - use '-nowf' to disable wireframing completely.
06/08/2019 13:14:26    - use '-nowcr' to disable the Copy Rectangle after the
06/08/2019 13:14:26      moved window is released in the new position.
06/08/2019 13:14:26   Also see the -help entry for tuning parameters.
06/08/2019 13:14:26   You can press 3 Alt_L's (Left "Alt" key) in a row to 
06/08/2019 13:14:26   repaint the screen, also see the -fixscreen option for
06/08/2019 13:14:26   periodic repaints.
06/08/2019 13:14:26 
06/08/2019 13:14:26 XFIXES available on display, resetting cursor mode
06/08/2019 13:14:26   to: '-cursor most'.
06/08/2019 13:14:26   to disable this behavior use: '-cursor arrow'
06/08/2019 13:14:26   or '-noxfixes'.
06/08/2019 13:14:26 using XFIXES for cursor drawing.
06/08/2019 13:14:26 GrabServer control via XTEST.
06/08/2019 13:14:26 
06/08/2019 13:14:26 Scroll Detection: -scrollcopyrect mode is in effect to
06/08/2019 13:14:26   use RECORD extension to try to detect scrolling windows
06/08/2019 13:14:26   (induced by either user keystroke or mouse input).
06/08/2019 13:14:26   If this yields undesired behavior (poor response, painting
06/08/2019 13:14:26   errors, etc) it may be disabled via: '-noscr'
06/08/2019 13:14:26   Also see the -help entry for tuning parameters.
06/08/2019 13:14:26   You can press 3 Alt_L's (Left "Alt" key) in a row to 
06/08/2019 13:14:26   repaint the screen, also see the -fixscreen option for
06/08/2019 13:14:26   periodic repaints.
06/08/2019 13:14:26 
06/08/2019 13:14:26 XKEYBOARD: number of keysyms per keycode 7 is greater
06/08/2019 13:14:26   than 4 and 195 keysyms are mapped above 4.
06/08/2019 13:14:26   Automatically switching to -xkb mode.
06/08/2019 13:14:26   If this makes the key mapping worse you can
06/08/2019 13:14:26   disable it with the "-noxkb" option.
06/08/2019 13:14:26   Also, remember "-remap DEAD" for accenting characters.
06/08/2019 13:14:26 
06/08/2019 13:14:26 X FBPM extension not supported.
06/08/2019 13:14:26 X display is capable of DPMS.
06/08/2019 13:14:26 --------------------------------------------------------
06/08/2019 13:14:26 
06/08/2019 13:14:26 initialize_xdisplay_fb: vis_str set to: TrueColor:32

Visual Info:
 set_visual("TrueColor:32")
 visual_depth: 32
 visual: 4
 initialize_xdisplay_fb()
 Visual*:    0x55d2ff3d1a20
 visualid:   0x63
 screen:     0
 depth:      32
 class:      4
 red_mask:   0x00ff0000  00000000111111110000000000000000
 green_mask: 0x0000ff00  00000000000000001111111100000000
 blue_mask:  0x000000ff  00000000000000000000000011111111
 cmap_size:  256
 bits b/rgb: 8

06/08/2019 13:14:26 Default visual ID: 0x63
06/08/2019 13:14:26 initialize_screen: fb_depth/fb_bpp/fb_Bpl 32/32/5120
06/08/2019 13:14:26 
06/08/2019 13:14:26 X display :7.0 is 32bpp depth=32 true color
06/08/2019 13:14:26 
06/08/2019 13:14:26 Autoprobing TCP port 
06/08/2019 13:14:26 Autoprobing selected TCP port 5900
06/08/2019 13:14:26 Autoprobing TCP6 port 
06/08/2019 13:14:26 Autoprobing selected TCP6 port 5900
06/08/2019 13:14:26 listen6: bind: Address already in use
06/08/2019 13:14:26 Not listening on IPv6 interface.
06/08/2019 13:14:26 
06/08/2019 13:14:26 Xinerama is present and active (e.g. multi-head).
06/08/2019 13:14:26 Xinerama: number of sub-screens: 1
06/08/2019 13:14:26 Xinerama: no blackouts needed (only one sub-screen)
06/08/2019 13:14:26 
06/08/2019 13:14:26 fb read rate: 77 MB/sec
06/08/2019 13:14:26 The X server says there are 10 mouse buttons.
06/08/2019 13:14:26 screen setup finished.
06/08/2019 13:14:26 
06/08/2019 13:14:26 WARNING: You are running x11vnc WITHOUT a password.  See
06/08/2019 13:14:26 WARNING: the warning message printed above for more info.
06/08/2019 13:14:26 

The VNC desktop is:      ltsp230:0

******************************************************************************
Have you tried the x11vnc '-ncache' VNC client-side pixel caching feature yet?

The scheme stores pixel data offscreen on the VNC viewer side for faster
retrieval.  It should work with any VNC viewer.  Try it by running:

    x11vnc -ncache 10 ...

One can also add -ncache_cr for smooth 'copyrect' window motion.
More info: http://www.karlrunge.com/x11vnc/faq.html#faq-client-caching

06/08/2019 13:14:26 Making connection to client on host server port 58979
06/08/2019 13:14:26   other clients:
06/08/2019 13:14:26 Normal socket connection
06/08/2019 13:14:26 Disabled X server key autorepeat.
06/08/2019 13:14:26   to force back on run: 'xset r on' (3 times)
06/08/2019 13:14:26 incr accepted_client=1 for 200.189.61.38:58979  sock=14
06/08/2019 13:14:26 reverse_connect: server:58979/200.189.61.38 OK
06/08/2019 13:14:26 reverse_connect: turning on auth for 200.189.61.38
06/08/2019 13:14:26 Client Protocol Version 3.8
06/08/2019 13:14:26 Protocol version sent 3.8, using 3.8
06/08/2019 13:14:26 client progressed=1 in 0/4 0.001032 s
06/08/2019 13:14:26 rfbProcessClientSecurityType: executing handler for type 1
06/08/2019 13:14:26 rfbProcessClientSecurityType: returning securityResult for client rfb version >= 3.8
06/08/2019 13:14:26 Pixel format for client 200.189.61.38:
06/08/2019 13:14:26   8 bpp, depth 6
06/08/2019 13:14:26   true colour: max r 3 g 3 b 3, shift r 4 g 2 b 0
06/08/2019 13:14:26 Enabling full-color cursor updates for client 200.189.61.38
06/08/2019 13:14:26 Enabling NewFBSize protocol extension for client 200.189.61.38
06/08/2019 13:14:26 Using ZRLE encoding for client 200.189.61.38
06/08/2019 13:14:27 created   xdamage object: 0x2600002
06/08/2019 13:14:27 client useCopyRect: 200.189.61.38 -1
06/08/2019 13:14:27 client_set_net: 200.189.61.38  0.0001
06/08/2019 13:14:27 copy_tiles: allocating first_line at size 41
06/08/2019 13:14:28 Pixel format for client 200.189.61.38:
06/08/2019 13:14:28   32 bpp, depth 24, little endian
06/08/2019 13:14:28   true colour: max r 255 g 255 b 255, shift r 16 g 8 b 0
06/08/2019 13:14:28 Enabling full-color cursor updates for client 200.189.61.38
06/08/2019 13:14:28 Enabling NewFBSize protocol extension for client 200.189.61.38
06/08/2019 13:14:28 Switching from ZRLE to hextile Encoding for client 200.189.61.38
06/08/2019 13:14:29 client 1 network rate 77.3 KB/sec (1427.2 eff KB/sec)
06/08/2019 13:14:29 client 1 latency:  100.5 ms
06/08/2019 13:14:29 dt1: 0.0737, dt2: 0.8952 dt3: 0.1005 bytes: 71049
06/08/2019 13:14:29 link_rate: LR_BROADBAND - 100 ms, 77 KB/s
06/08/2019 13:14:35 created selwin: 0x2600003
06/08/2019 13:14:35 called initialize_xfixes()
caught XIO error:
06/08/2019 13:14:40 deleted 40 tile_row polling images.

=======================================

Answer Sever

Tue Aug  6 13:14:26 2019
 CConn:       Accepted connection from 0.0.0.0::48912
 CConnection: Server supports RFB protocol version 3.8
 CConnection: Using RFB protocol version 3.8
 TXImage:     Using default colormap and visual, TrueColor, depth 24.
 CConn:       Using pixel format depth 6 (8bpp) rgb222
 CConn:       Using ZRLE encoding

Tue Aug  6 13:14:28 2019
 CConn:       Throughput 20000 kbit/s - changing to hextile encoding
 CConn:       Throughput 20000 kbit/s - changing to full colour
 CConn:       Using pixel format depth 24 (32bpp) little-endian rgb888
 CConn:       Using hextile encoding

Tue Aug  6 13:14:40 2019
 main:        read: Connection reset by peer (104)

**Comment by alkisg** [comment]
So this happens only when you try to monitor the client?
If so, it sounds like an issue with x11vnc, not epoptes...

**Comment by PedroCoelhho** [comment]
would you have any suggestions on how to proceed?

**Comment by alkisg** [comment]
On the server, close epoptes, open a  terminal, and run:

```shell
xvnc4viewer -listen
```

On the client, run:
```shell
x11vnc -connect server
```

If this crashes too, file a bug report in x11vnc telling them the errors you see from this minimal test.

By the way, what's the output of this command on the server?
```shell
dpkg-query -W x11vnc
```

**Comment by PedroCoelhho** [comment]
Server

user@ltsp-admin:~$ xvnc4viewer -listen

VNC Viewer Free Edition 4.1.1 for X - built Feb 25 2015 22:57:51
Copyright (C) 2002-2005 RealVNC Ltd.
See http://www.realvnc.com for information on VNC.

Wed Aug  7 13:53:07 2019
 main:        Listening on port 5500

  
Wed Aug  7 13:56:30 2019
 CConn:       Accepted connection from 0.0.0.0::39258
 CConnection: Server supports RFB protocol version 3.8
 CConnection: Using RFB protocol version 3.8

Wed Aug  7 13:56:31 2019
 TXImage:     Using default colormap and visual, TrueColor, depth 24.
 CConn:       Using pixel format depth 6 (8bpp) rgb222
 CConn:       Using ZRLE encoding

Wed Aug  7 13:56:37 2019
 CConn:       Throughput 20142 kbit/s - changing to hextile encoding
 CConn:       Throughput 20142 kbit/s - changing to full colour
 CConn:       Using pixel format depth 24 (32bpp) little-endian rgb888
 CConn:       Using hextile encoding

Wed Aug  7 13:56:43 2019
 CConn:       Throughput 20000 kbit/s - changing to raw encoding
 CConn:       Using raw encoding

=================================================================
Client

atendimento7@ltsp230:~$ x11vnc -connect 200.189.61.38
###############################################################
#@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@#
#@                                                           @#
#@  **  WARNING  **  WARNING  **  WARNING  **  WARNING  **   @#
#@                                                           @#
#@        YOU ARE RUNNING X11VNC WITHOUT A PASSWORD!!        @#
#@                                                           @#
#@  This means anyone with network access to this computer   @#
#@  may be able to view and control your desktop.            @#
#@                                                           @#
#@ >>> If you did not mean to do this Press CTRL-C now!! <<< @#
#@                                                           @#
#@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@#
#@                                                           @#
#@  You can create an x11vnc password file by running:       @#
#@                                                           @#
#@       x11vnc -storepasswd password /path/to/passfile      @#
#@  or   x11vnc -storepasswd /path/to/passfile               @#
#@  or   x11vnc -storepasswd                                 @#
#@                                                           @#
#@  (the last one will use ~/.vnc/passwd)                    @#
#@                                                           @#
#@  and then starting x11vnc via:                            @#
#@                                                           @#
#@      x11vnc -rfbauth /path/to/passfile                    @#
#@                                                           @#
#@  an existing ~/.vnc/passwd file from another VNC          @#
#@  application will work fine too.                          @#
#@                                                           @#
#@  You can also use the -passwdfile or -passwd options.     @#
#@  (note -passwd is unsafe if local users are not trusted)  @#
#@                                                           @#
#@  Make sure any -rfbauth and -passwdfile password files    @#
#@  cannot be read by untrusted users.                       @#
#@                                                           @#
#@  Use x11vnc -usepw to automatically use your              @#
#@  ~/.vnc/passwd or ~/.vnc/passwdfile password files.       @#
#@  (and prompt you to create ~/.vnc/passwd if neither       @#
#@  file exists.)  Under -usepw, x11vnc will exit if it      @#
#@  cannot find a password to use.                           @#
#@                                                           @#
#@                                                           @#
#@  Even with a password, the subsequent VNC traffic is      @#
#@  sent in the clear.  Consider tunnelling via ssh(1):      @#
#@                                                           @#
#@    http://www.karlrunge.com/x11vnc/#tunnelling            @#
#@                                                           @#
#@  Or using the x11vnc SSL options: -ssl and -stunnel       @#
#@                                                           @#
#@  Please Read the documention for more info about          @#
#@  passwords, security, and encryption.                     @#
#@                                                           @#
#@    http://www.karlrunge.com/x11vnc/faq.html#faq-passwd    @#
#@                                                           @#
#@  To disable this warning use the -nopw option, or put     @#
#@  'nopw' on a line in your ~/.x11vncrc file.               @#
#@                                                           @#
#@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@#
###############################################################
07/08/2019 13:56:29 x11vnc version: 0.9.13 lastmod: 2011-08-10  pid: 5380
07/08/2019 13:56:29 Using X display :7.0
07/08/2019 13:56:29 rootwin: 0x106 reswin: 0x2600001 dpy: 0x716efc00
07/08/2019 13:56:29 
07/08/2019 13:56:29 ------------------ USEFUL INFORMATION ------------------
07/08/2019 13:56:29 X DAMAGE available on display, using it for polling hints.
07/08/2019 13:56:29   To disable this behavior use: '-noxdamage'
07/08/2019 13:56:29 
07/08/2019 13:56:29   Most compositing window managers like 'compiz' or 'beryl'
07/08/2019 13:56:29   cause X DAMAGE to fail, and so you may not see any screen
07/08/2019 13:56:29   updates via VNC.  Either disable 'compiz' (recommended) or
07/08/2019 13:56:29   supply the x11vnc '-noxdamage' command line option.
07/08/2019 13:56:29 
07/08/2019 13:56:29 Wireframing: -wireframe mode is in effect for window moves.
07/08/2019 13:56:29   If this yields undesired behavior (poor response, painting
07/08/2019 13:56:29   errors, etc) it may be disabled:
07/08/2019 13:56:29    - use '-nowf' to disable wireframing completely.
07/08/2019 13:56:29    - use '-nowcr' to disable the Copy Rectangle after the
07/08/2019 13:56:29      moved window is released in the new position.
07/08/2019 13:56:29   Also see the -help entry for tuning parameters.
07/08/2019 13:56:29   You can press 3 Alt_L's (Left "Alt" key) in a row to 
07/08/2019 13:56:29   repaint the screen, also see the -fixscreen option for
07/08/2019 13:56:29   periodic repaints.
07/08/2019 13:56:29 
07/08/2019 13:56:29 XFIXES available on display, resetting cursor mode
07/08/2019 13:56:29   to: '-cursor most'.
07/08/2019 13:56:29   to disable this behavior use: '-cursor arrow'
07/08/2019 13:56:29   or '-noxfixes'.
07/08/2019 13:56:29 using XFIXES for cursor drawing.
07/08/2019 13:56:29 GrabServer control via XTEST.
07/08/2019 13:56:29 
07/08/2019 13:56:29 Scroll Detection: -scrollcopyrect mode is in effect to
07/08/2019 13:56:29   use RECORD extension to try to detect scrolling windows
07/08/2019 13:56:29   (induced by either user keystroke or mouse input).
07/08/2019 13:56:29   If this yields undesired behavior (poor response, painting
07/08/2019 13:56:29   errors, etc) it may be disabled via: '-noscr'
07/08/2019 13:56:29   Also see the -help entry for tuning parameters.
07/08/2019 13:56:29   You can press 3 Alt_L's (Left "Alt" key) in a row to 
07/08/2019 13:56:29   repaint the screen, also see the -fixscreen option for
07/08/2019 13:56:29   periodic repaints.
07/08/2019 13:56:29 
07/08/2019 13:56:29 XKEYBOARD: number of keysyms per keycode 7 is greater
07/08/2019 13:56:29   than 4 and 195 keysyms are mapped above 4.
07/08/2019 13:56:29   Automatically switching to -xkb mode.
07/08/2019 13:56:29   If this makes the key mapping worse you can
07/08/2019 13:56:29   disable it with the "-noxkb" option.
07/08/2019 13:56:29   Also, remember "-remap DEAD" for accenting characters.
07/08/2019 13:56:29 
07/08/2019 13:56:29 X FBPM extension not supported.
07/08/2019 13:56:29 X display is capable of DPMS.
07/08/2019 13:56:29 --------------------------------------------------------
07/08/2019 13:56:29 
07/08/2019 13:56:29 Default visual ID: 0x21
07/08/2019 13:56:29 Read initial data from X display into framebuffer.
07/08/2019 13:56:29 initialize_screen: fb_depth/fb_bpp/fb_Bpl 24/32/5120
07/08/2019 13:56:29 
07/08/2019 13:56:29 X display :7.0 is 32bpp depth=24 true color
07/08/2019 13:56:29 
07/08/2019 13:56:29 Autoprobing TCP port 
07/08/2019 13:56:29 Autoprobing selected TCP port 5900
07/08/2019 13:56:29 Autoprobing TCP6 port 
07/08/2019 13:56:29 Autoprobing selected TCP6 port 5900
07/08/2019 13:56:29 listen6: bind: Address already in use
07/08/2019 13:56:29 Not listening on IPv6 interface.
07/08/2019 13:56:29 
07/08/2019 13:56:29 Xinerama is present and active (e.g. multi-head).
07/08/2019 13:56:29 Xinerama: number of sub-screens: 1
07/08/2019 13:56:29 Xinerama: no blackouts needed (only one sub-screen)
07/08/2019 13:56:29 
07/08/2019 13:56:29 fb read rate: 151 MB/sec
07/08/2019 13:56:29 fast read: reset -wait  ms to: 10
07/08/2019 13:56:29 fast read: reset -defer ms to: 10
07/08/2019 13:56:29 The X server says there are 10 mouse buttons.
07/08/2019 13:56:29 screen setup finished.
07/08/2019 13:56:29 
07/08/2019 13:56:29 WARNING: You are running x11vnc WITHOUT a password.  See
07/08/2019 13:56:29 WARNING: the warning message printed above for more info.
07/08/2019 13:56:29 

The VNC desktop is:      ltsp230:0
PORT=5900

******************************************************************************
Have you tried the x11vnc '-ncache' VNC client-side pixel caching feature yet?

The scheme stores pixel data offscreen on the VNC viewer side for faster
retrieval.  It should work with any VNC viewer.  Try it by running:

    x11vnc -ncache 10 ...

One can also add -ncache_cr for smooth 'copyrect' window motion.
More info: http://www.karlrunge.com/x11vnc/faq.html#faq-client-caching

07/08/2019 13:56:30 Making connection to client on host 200.189.61.38 port 5500
07/08/2019 13:56:30   other clients:
07/08/2019 13:56:30 Normal socket connection
07/08/2019 13:56:30 Disabled X server key autorepeat.
07/08/2019 13:56:30   to force back on run: 'xset r on' (3 times)
07/08/2019 13:56:30 incr accepted_client=1 for 200.189.61.38:5500  sock=11
07/08/2019 13:56:30 reverse_connect: 200.189.61.38/200.189.61.38 OK
07/08/2019 13:56:30 reverse_connect: turning on auth for 200.189.61.38
07/08/2019 13:56:30 client progressed=0 in 15/10 0.015308 s
07/08/2019 13:56:30 Client Protocol Version 3.8
07/08/2019 13:56:30 Protocol version sent 3.8, using 3.8
07/08/2019 13:56:30 rfbProcessClientSecurityType: executing handler for type 1
07/08/2019 13:56:30 rfbProcessClientSecurityType: returning securityResult for client rfb version >= 3.8
07/08/2019 13:56:30 created   xdamage object: 0x260002c
07/08/2019 13:56:30 copy_tiles: allocating first_line at size 41
07/08/2019 13:56:31 client useCopyRect: 200.189.61.38 0
07/08/2019 13:56:31 client_set_net: 200.189.61.38  0.0020
07/08/2019 13:56:31 Pixel format for client 200.189.61.38:
07/08/2019 13:56:31   8 bpp, depth 6
07/08/2019 13:56:31   true colour: max r 3 g 3 b 3, shift r 4 g 2 b 0
07/08/2019 13:56:31 Enabling full-color cursor updates for client 200.189.61.38
07/08/2019 13:56:31 Enabling NewFBSize protocol extension for client 200.189.61.38
07/08/2019 13:56:31 Using ZRLE encoding for client 200.189.61.38
07/08/2019 13:56:36 cursor_noshape_updates_clients: 0
07/08/2019 13:56:37 Pixel format for client 200.189.61.38:
07/08/2019 13:56:37   32 bpp, depth 24, little endian
07/08/2019 13:56:37   true colour: max r 255 g 255 b 255, shift r 16 g 8 b 0
07/08/2019 13:56:37 no translation needed
07/08/2019 13:56:37 Enabling full-color cursor updates for client 200.189.61.38
07/08/2019 13:56:37 Enabling NewFBSize protocol extension for client 200.189.61.38
07/08/2019 13:56:37 Switching from ZRLE to hextile Encoding for client 200.189.61.38
07/08/2019 13:56:37 cursor_noshape_updates_clients: 0
07/08/2019 13:56:42 cursor_noshape_updates_clients: 0
07/08/2019 13:56:42 created selwin: 0x260002d
07/08/2019 13:56:42 called initialize_xfixes()
07/08/2019 13:56:42 cursor_noshape_updates_clients: 0
07/08/2019 13:56:43 Enabling full-color cursor updates for client 200.189.61.38
07/08/2019 13:56:43 Enabling NewFBSize protocol extension for client 200.189.61.38
07/08/2019 13:56:43 Switching from hextile to raw Encoding for client 200.189.61.38
07/08/2019 13:56:45 client 1 network rate 10433.6 KB/sec (10433.6 eff KB/sec)
07/08/2019 13:56:45 client 1 latency:  6.3 ms
07/08/2019 13:56:45 dt1: 0.4599, dt2: 0.0458 dt3: 0.0063 bytes: 5242892
07/08/2019 13:56:45 link_rate: LR_LAN - 6 ms, 10433 KB/s
07/08/2019 13:57:08 cursor_noshape_updates_clients: 0
07/08/2019 13:57:15 rfbProcessClientNormalMessage: read: Connection reset by peer
07/08/2019 13:57:15 client_count: 0
07/08/2019 13:57:15 Restored X server key autorepeat to: 1
07/08/2019 13:57:15 viewer exited.
07/08/2019 13:57:15 deleted 40 tile_row polling images.


=================================================================
user@ltsp-admin:~$ dpkg-query -W x11vnc
x11vnc	0.9.13-6

**Comment by alkisg** [comment]
To sum up:

I knew that x11vnc 0.9.13-3 had [issues](https://bugs.launchpad.net/ubuntu/+source/x11vnc/+bug/1686084), so I copied the newer x11vnc 0.9.13-6 in the Greek schools PPA.
You have the fixed one though.
The problem happens without epoptes, by just running xvnc4viewer on the server and x11vnc on a client.

So, the problem is in x11vnc; file a bug report against that e.g. in [launchpad](https://launchpad.net/ubuntu/+source/x11vnc/+bugs) or elsewhere.
I'm closing this issue as it cannot be fixed from the Epoptes code.

Also note that if you have a firewall, it might affect things, try to temporarily disable it.
